### PR TITLE
objc: rename IMP to NewIMP and _IMP to IMP

### LINKED ISSUE #22: objc: `_IMP` is not exported though this is used in exported functions
I think we should rename `_IMP` to an exported one. @TotallyGamerJet what do you think?

### PR DESCRIPTION
**Pull request by hajimehoshi**
Closes #22

### CODE REVIEW

File: objc/objc_runtime_darwin.go [hunk 1]
```diff
@@ -215,19 +215,18 @@ func (i Ivar) Offset() uintptr {
...
 
-// IMP takes a Go function that takes (ID, SEL) as its first two arguments. It returns an _IMP function
+// NewIMP takes a Go function that takes (ID, SEL) as its first two arguments. It returns an IMP function
 // pointer that can be called by Objective-C code. The function pointer is never deallocated.
-func IMP(fn interface{}) _IMP {
+func NewIMP(fn interface{}) IMP {
 	// this is only here so that it is easier to port C code to Go.
 	// this is not guaranteed to be here forever so make sure to port your callbacks to Go
 	// If you have a C function pointer cast it to a uintptr before passing it
 	// to this function.
```
Comment (TotallyGamerJet) [inline objc/objc_runtime_darwin.go:227]: Do we want to make this part of the exported API? Or just leave it as internal

Comment (hajimehoshi) [reply objc/objc_runtime_darwin.go:227]: I think we should make it exported, or how do users use this?

Comment (TotallyGamerJet) [reply objc/objc_runtime_darwin.go:227]: Or do we not care since you can now cast a uintptr to IMP?

Comment (TotallyGamerJet) [reply objc/objc_runtime_darwin.go:227]: It would be used like `NewIMP(uintptr(C.someCFunction))`

Comment (hajimehoshi) [reply objc/objc_runtime_darwin.go:227]: Let's go back to discuss the idea https://github.com/ebitengine/purego/issues/22

Comment (hajimehoshi) [reply objc/objc_runtime_darwin.go]: Updated: I have removed uintptr usage.

**Review by TotallyGamerJet — APPROVED** [review]
LGTM!

**Comment by hajimehoshi** [comment]
Oh, I wanted to edit the commit message before merging...

**Comment by hajimehoshi** [comment]
Thank you for reviewing!

**Comment by TotallyGamerJet** [comment]
Whoops! My bad

**Pull request merged**